core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -3847,29 +3864,16 @@ private void o3ProcessPartitionRemoveCandidates0(int n) {
                     );
                     TableUtils.txnPartitionConditionally(other, txn);
                     other.$();
-                    if (ff.isSoftLink(other)) {
-                        // in windows ^ ^ will return false, but that is ok as the behaviour
-                        // is to delete the link, not the contents of the target. in *nix
-                        // systems we can simply unlink, which deletes the link and leaves
-                        // the contents of the target intact
-                        if (ff.unlink(other) == 0) {
-                            LOG.info().$("purged by unlink [path=").$(other).I$();
-                            return;
-                        } else {
-                            LOG.error().$("failed to unlink, will delete [path=").$(other).I$();
+                    if (!txWriter.isPartitionReadOnlyByPartitionTimestamp(timestamp)) {
```

Comment:
Here the partition is already removed from the table, too late to check.
Remove the check from here or leave it as an assertion only.

Reply:
does this still apply?

Reply:
Does this comment still hold? I've checked the code and I don't understand why "the partition is already removed from the table, too late to check" applies. Probably I'm missing something.